Add M22 SD release for onboard cards

cptwobbles opened the pull request:
### Description

There was no menu item to release onboard SD card for USB access. I was having to resort to cycling power to release it.

This only adds the `M22` "Release SD card" menu item when `USB_SD_ONBOARD` is enabled.

Replaces `M21` menu item with `M22`. I don't think missing the `M21` will be a big concern as you can easily release the SD and initialize it again.

This is my first time using GitHub so please don't hesitate crucifying me for any wrong doings. Hang on, I think I've stuffed up already, Sorry about the double up, I think I've got it right this time...

### Benefits

Speeds up file transfer to the printer without removing the SD card or sending a serial `M22`.

### Related Issues

I have tested this code on my SKR 1.3 and it works great but as I mentioned, someone might not like losing the `M21` if they are manually removing the SD card due to an extra button press..

cptwobbles committed:
Use M22 Release for reader without SD_DETECT_PIN

gloomyandy commented:
Previous discussion here: https://github.com/MarlinFirmware/Marlin/issues/11809

I'm not not sure that USB_SD_ONBOARD is the correct test to apply though. As this can still be defined even if users are using the LCD based reader for Marlin. If you are trying to only do this when sharing and SD card it should probably be testing both USB_SD_ONBOARD and LPC_SD_ONBOARD. 

Personally I think that this code should jut be the standard case, as it should really always be possible to release the SD card in a clean way, especially if there is no SD detect pin.

cptwobbles commented:
Thanks for that, your're right about needing both USB_SD_ONBOARD and LPC_SD_ONBOARD. I had a read of  #11809 and the conclusion to replace "Change SD" with "Release SD" / "Insert SD" looks good. I'll have a rethink of the code.

cptwobbles commented:
I've updated it to function as discussed in #11809. M22 "Release SD card" instead of M21 "Change SD card" but only when no detect pin. Functions as normal with a detect pin.

Added MSG_SD_RELEASED "Card released" as it is nicer than the MSG_NO_CARD "No SD card".

Also thinking that MSG_SD_RELEASED could say something like "Safe to remove card".
